### PR TITLE
feat: back out Connect auto-import feature

### PR DESCRIPTION
**Pull request by agavra**
### Description 

After discussions with @derekjn and @MichaelDrogalis we think it's probably better to back out the connect auto-import. Instead, describing a supported connector will list the topics available (see unit tests).

### Testing done 

- Added some unit tests
- E2E Local Testing

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Comment by big-andy-coates** [comment]
Boo!

What is our approach to actually providing this as an integrated solution then?  We are planning on having proper auto-import right? Otherwise I'd say it will still feel like too products and not the integrated solution we were targeting.

File: ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java [hunk 1]
```diff
@@ -648,12 +648,12 @@ private void printConnectorDescription(final ConnectorDescription description) {
     writer().println(String.format("%-20s : %s", "Type", description.getStatus().type()));
     writer().println(String.format("%-20s : %s", "State", status.connector().state()));
     writer().println(String.format("%-20s : %s", "WorkerId", status.connector().workerId()));
-    if (!status.connector().trace().isEmpty()) {
+    if (status.connector().trace() != null && !status.connector().trace().isEmpty()) {
```
Comment (big-andy-coates) [inline ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java]: Yuck :(

Can we raise an upstream change to ensure this is never null, but an empty string?

```suggestion
    if (!ObjectUtils.defaultIfNull(status.connector().trace(), "").isEmpty()) {
```

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/DescribeConnectorExecutor.java [hunk 2]
```diff
@@ -106,13 +113,39 @@ public DescribeConnectorExecutor() {
...
+    }
+
     final ConnectorDescription description = new ConnectorDescription(
         configuredStatement.getStatementText(),
         info.config().get(ConnectorConfig.CONNECTOR_CLASS_CONFIG),
         status,
-        sources
+        sources,
+        topics,
+        warnings
```
Comment (big-andy-coates) [inline ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/DescribeConnectorExecutor.java]: Looks like `ConnectorDescription` is only ever created with _either_ the topics list _or_ a single warning.

Might be worth codifying this. i.e. the `topics` param passed in to `ConnectorDescription` should be a union type of _either_ the topic list _or_ a single warning.

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/DescribeConnectorExecutor.java [hunk 3]
```diff
@@ -106,13 +113,39 @@ public DescribeConnectorExecutor() {
...
     return Optional.of(description);
   }
+
+  private List<String> getTopics(final Admin admin, final Optional<Connector> connector)
+      throws Exception {
+    if (!connector.isPresent()) {
+      return ImmutableList.of();
+    }
+
+    return admin.listTopics().names().get(5, TimeUnit.SECONDS)
```
Comment (big-andy-coates) [inline ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/DescribeConnectorExecutor.java]: Consider making this timeout configurable?

**Comment by MichaelDrogalis** [comment]
> Boo!
> 
> What is our approach to actually providing this as an integrated solution then? We are planning on having proper auto-import right? Otherwise I'd say it will still feel like too products and not the integrated solution we were targeting.

We need to figure out how to do this without providing special support in KSQL for particular connectors. We don't want to be in the business of having to implement particular interfaces for every connector out there and always playing catch-up.

**Comment by agavra** [comment]
Building off of what @MichaelDrogalis said, it's not just per-connector things (even after this revert we still have per-connector things that we'll need to remove) it's that the experience around auto-import isn't great and extremely limited, and the overhead of typing one `CREATE` statement (without a schema, thanks to SR) isn't that high.

File: ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java (hunk 1 above)
Comment (agavra) [reply ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java]: yeah I was surprised about that as well, not sure how I didn't catch it till now! if I have time I'll look into changing it upstream, but public API changes in AK...

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/DescribeConnectorExecutor.java (hunk 2 above)
Comment (agavra) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/DescribeConnectorExecutor.java]: In the future we might want to supply other warnings that are unrelated to the list of topics (e.g. failed to connect to Connect) - I don't think that coupling necessarily makes sense.

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/DescribeConnectorExecutor.java (hunk 3 above)
Comment (agavra) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/DescribeConnectorExecutor.java]: haha calling me out on indolence.

Comment (agavra) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/DescribeConnectorExecutor.java]: So I looked into this a little bit, we don't have any configureable timeouts at the moment (in KsqlConfig anyway). I'm going to follow this PR with one that adds a `DEFAULT_EXTERNAL_SERVICE_TIMEOUT` and then we can add specific overrides for it.

**Pull request merged**